stl/inc/xutility

```diff
@@ -4917,7 +4922,13 @@ template <class _CtgIt, class _Ty>
 void _Fill_memset(_CtgIt _Dest, const _Ty _Val, const size_t _Count) {
     // implicitly convert (a cast would suppress warnings); also handles _Iter_value_t<_CtgIt> being bool
     _Iter_value_t<_CtgIt> _Dest_val = _Val;
-    _CSTD memset(_To_address(_Dest), static_cast<unsigned char>(_Dest_val), _Count);
+    if constexpr (sizeof(_Dest_val) == 1) {
+        _CSTD memset(_To_address(_Dest), _Bit_cast<unsigned char>(_Dest_val), _Count);
+    } else if constexpr (sizeof(_Dest_val) == 2) {
+        _CSTD wmemset(reinterpret_cast<wchar_t*>(_To_address(_Dest)), _Bit_cast<unsigned short>(_Dest_val), _Count);

```

Comment:
I'm not sure whether UCRT's `wmemset` can handle unaligned write or not. Perhaps it would be safer to perform manual aligning for types whose alignment is only 1, or to restrict this branch for types whose size and alignment are both 2.

Reply:
I'm not sure...

On platforms that are alignment sensitive, the usual `std::fill` may crash as well.

You'll have to add `__unaligned` to the pointer types to work with such pointers, and there's no guarantee the STL propagates that modifier correctly to its algorithms, as it is non-goal to support non-standard extensions,

Reply:
Oh, I see what you mean. Perhaps yes...

Reply:
Fixed